agent.go

```diff
@@ -786,21 +934,80 @@ func grpcTracer(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo
 	return resp, err
 }
 
+func makeUnaryInterceptor() grpc.UnaryServerInterceptor {
+	return func(origCtx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		grpcCall := info.FullMethod
+
+		span, _ := trace(getGRPCContext(), "gRPC", grpcCall)
+		span.SetTag("grpc-method-type", "unary")
+		defer span.Finish()
+
+		if strings.HasSuffix(grpcCall, "/ReadStdout") || strings.HasSuffix(grpcCall, "/WriteStdin") {
+			// Add a tag to allow filtering of those calls dealing
+			// input and output. These tend to be very long and
+			// being able to filter them out allows the
+			// performance of "core" calls to be determined
+			// without the "noise" of these calls.
+			span.SetTag("api-category", "interactive")
+		}
+
+		// Use the context which will provide the correct trace
+		// ordering, *NOT* the context provided to the function
+		// returned by this function.
+		return handler(getGRPCContext(), req)
+	}
+}
+
+// makeStreamInterceptor creates a function to handle tracing of stream-based gRPC calls.
+// The reason this needs to return a function being that when this function is
+// called, the correct context to use for tracing is not available so the
+// returned function must query it when called later.
+func makeStreamInterceptor() grpc.StreamServerInterceptor {
```

Comment:
@jodh-intel @sboeuf do we actually do streaming gRPC today.

Reply:
I don't think so, but I added this to "cover all bases" ;)